sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -285,6 +286,116 @@ case class ArrayTransform(
   override def prettyName: String = "transform"
 }
 
+/**
+ * Sorts elements in an array using a comparator function.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(expr, func) - Sorts the input array in ascending order. The elements of the " +
+    "input array must be orderable. Null elements will be placed at the end of the returned " +
+    "array. Since 3.0.0 also sorts and returns the array based on the given " +
+    "comparator function. The comparator will take two nullable arguments " +
+    "representing two nullable elements of the array." +
+    "It returns -1, 0, or 1 as the first nullable element is less than, equal to, or greater " +
+    "than the second nullable element. If the comparator function returns other " +
+    "values (including NULL), the query will fail and raise an error.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(array(5, 6, 1), (left, right) ->
+        If(And(IsNull(left), IsNull(right)), 0,
+          If(IsNull(left), 1, If(IsNull(right), -1,
+           If(LessThan(left, right), -1, If(GreaterThan(left, right), 1, 0)))));
+       [1,5,6]
+      > SELECT _FUNC_(array('bc', 'ab', 'dc'), (left, right) ->
+        If(And(IsNull(left), IsNull(right)), 0,
+          If(IsNull(left), 1, If(IsNull(right), -1,
+           If(LessThan(left, right), -1, If(GreaterThan(left, right), 1, 0)))));
+       ["dc", "bc", "ab"]
```

Comment:
Seems like these commands are not checked. We need to fit in one line. We can use `// scalastyle:off line.size.limit` if the command is not fit in one line.
E.g., see `ZipWith`.